test/test_eth_registrar_controller.js

```diff
@@ -58,150 +61,192 @@ contract('ETHRegistrarController', function (accounts) {
+		var balanceBefore = await web3.eth.getBalance(controller.address);
+		var tx = await controller.registerWithConfig("newconfigname2", registrantAccount, 28 * DAYS, secret, resolver.address, NULL_ADDRESS, {value: 28 * DAYS + 1, gasPrice: 0});
+		assert.equal(tx.logs.length, 1);
+		assert.equal(tx.logs[0].event, "NameRegistered");
+		assert.equal(tx.logs[0].args.name, "newconfigname2");
+		assert.equal(tx.logs[0].args.owner, registrantAccount);
+		assert.equal((await web3.eth.getBalance(controller.address)) - balanceBefore, 28 * DAYS);
+
+		var nodehash = namehash.hash("newconfigname2.eth");
+		assert.equal((await ens.resolver(nodehash)), resolver.address);
+		assert.equal((await resolver.addr(nodehash)), 0);
+	});
+
+	it('should include the owner in the commitment', async () => {
+		await controller.commit(await controller.makeCommitment("newname2", accounts[2], secret));
+
+		await advanceTime((await controller.minCommitmentAge()).toNumber());
+		var balanceBefore = await web3.eth.getBalance(controller.address);
+		await expectFailure(controller.register("newname2", registrantAccount, 28 * DAYS, secret, {value: 28 * DAYS, gasPrice: 0}));
+	});
+
+	it('should reject duplicate registrations', async () => {
+		await controller.commit(await controller.makeCommitment("newname", registrantAccount, secret));
+
+		await advanceTime((await controller.minCommitmentAge()).toNumber());
+		var balanceBefore = await web3.eth.getBalance(controller.address);
+		await expectFailure(controller.register("newname", registrantAccount, 28 * DAYS, secret, {value: 28 * DAYS, gasPrice: 0}));
+	});
+
+	it('should reject for expired commitments', async () => {
+		await controller.commit(await controller.makeCommitment("newname2", registrantAccount, secret));
+
+		await advanceTime((await controller.maxCommitmentAge()).toNumber() + 1);
+		var balanceBefore = await web3.eth.getBalance(controller.address);
+		await expectFailure(controller.register("newname2", registrantAccount, 28 * DAYS, secret, {value: 28 * DAYS, gasPrice: 0}));
+	});
+
+	it('should allow anyone to renew a name', async () => {
+		var expires = await baseRegistrar.nameExpires(sha3("newname"));
+		var balanceBefore = await web3.eth.getBalance(controller.address);
+		await controller.renew("newname", 86400, {value: 86400 + 1});
+		var newExpires = await baseRegistrar.nameExpires(sha3("newname"));
+		assert.equal(newExpires.toNumber() - expires.toNumber(), 86400);
+		assert.equal((await web3.eth.getBalance(controller.address)) - balanceBefore, 86400);
+	});
+
+	it('should require sufficient value for a renewal', async () => {
+		await expectFailure(controller.renew("newname", 86400));
+	});
+
+	it('should permit bulk renewal', async () => {
+		var balanceBefore = await web3.eth.getBalance(controller.address);
+		const nameExpires = await baseRegistrar.nameExpires(sha3("newname"));
+		await controller.renewAll(["newname", "newname"], 86400, NULL_ADDRESS, {value: 86400 * 2 + 1});
```

Comment:
Why is the test renewing same name (`newname`) twice?  Even if the effect is the same, better to renew different names to make the intention of the test clearler

Reply:
It makes it easier to check the outcome, since you only have to check one result.